### ISSUE TITLE
Add manual SQL Injection steps into A1 app

### ISSUE THREAD

**Issue by rafaveira3** [globocom/secDevLabs#81]
It would be great if `CopyNPaste` attack narrative includes also a manual attack to demonstrate a SQL injection exploitation.

**Comment by giancarlopro** [comment]
can i do this one?

**Comment by Krlier** [comment]
Definitely, @giancarlopro! We'd be very happy to review any PR you send us.

**Comment by Krlier** [comment]
> thanks, I will create an app for this issue ;)

Hey, @giancarlopro!

How's it going?

Can we help you with anything?

**Comment by giancarlopro** [comment]
> > thanks, I will create an app for this issue ;)
> 
> Hey, @giancarlopro!
> 
> How's it going?
> 
> Can we help you with anything?

Hi, i've been a bit busy lately and couldn't finish it yet, i thought it were vulnerable to normal sql injection, but it is vulnerable to blind sql injection and i found it more difficult to write about because of the repetitive nature of the method. So i thought about writing a script in python to automate it because doing it manually is a bit overkill. This is what i've writen so far

<hr />

For this attack narrative, we need a valid user, we will use userBianca with password 123456 , and we will use blind sql injection techniques to get sensitive information from the server.
The following two payloads can be used to test if the app is vulnerable to this type o attack.
```
Bianca' and 1=1 - 
Bianca' and 1=2 - 
```
If we get two different responses from those payloads, the app probably is vulnerable.

Now, since the app don't return the result from the query executed, we need to make assumptions about the data, and use the response to confirm those assumptions.
The following payload can be used to check if there's any table with the name starting with letter 'u':
```
Bianca' UNION SELECT table_name,2,3 FROM information_schema.tables WHERE SUBSTRING(table_name,1,1) = 'u' - 
```
If the app responds with User not found or wrong password! it means that there is a table starting with 'u', then we can change the second parameter ofSUBSTRING(table_name, 2, 1) to test the second character. And repeat those steps until we find the table name. The one we're looking for is Users .
Now we do the same to find the columns from table Users . Using the same payload with small changes we can test the column names.
```
Bianca' UNION SELECT column_name,2,3 FROM information_schema.columns WHERE SUBSTRING(column_name,1,1) = 'u' WHERE table_name='Users' - 
```
And again we repeat it and to get all the column names.

**Comment by Krlier** [comment]
Nice job, @giancarlopro! Excellent work so far! 🎉 

I understand the repetitive nature of this exploration method. Although, I believe it would be cool to be able to see the steps needed to exploit the app manually, which, in this case, does not mean we need to write every step needed.

In the scenario of this app, you could try describing what you intend to accomplish by using a certain command multiple times and, after those many times, what the app should return and what you'll do with it. Similar to what you did above!

What do you think?

If you'd like, add your suggested changes to the app's [README](https://github.com/globocom/secDevLabs/blob/master/owasp-top10-2017-apps/a1/copy-n-paste/README.md) and we can work together on writing the narrative! 🙂
